js/pulse.js

```diff
@@ -363,7 +364,9 @@
 
                 this.ready = true;
                 this._deviceContainer = adManagerMetadata.pulse_device_container;
-                this._persistentId = adManagerMetadata.pulse_persistent_id;
+                if (adManagerMetadata.pulse_persistent_id) {
+                  this._persistentId = adManagerMetadata.pulse_persistent_id;
```

Comment:
This will not change it if already present in the metadata, but `_onDeviceIdSet` doesn't have such a check. Shouldn't it be there as well? Or how is `DEVICE_ID_SET` triggered?